homeassistant/components/unifi/update.py

```diff
@@ -0,0 +1,128 @@
+    controller.entities[DOMAIN] = {DEVICE_UPDATE: set()}
+
+    @callback
+    def items_added(
+        clients: set = controller.api.clients, devices: set = controller.api.devices
+    ) -> None:
+        """Add device update entities."""
+        add_device_update_entities(controller, async_add_entities, devices)
+
+    for signal in (controller.signal_update, controller.signal_options_update):
+        config_entry.async_on_unload(
+            async_dispatcher_connect(hass, signal, items_added)
+        )
+
+    items_added()
+
+
+@callback
+def add_device_update_entities(controller, async_add_entities, devices):
+    """Add new device update entities from the controller."""
+    entities = []
+
+    for mac in devices:
+        if mac in controller.entities[DOMAIN][UniFiDeviceUpdateEntity.TYPE]:
+            continue
+
+        device = controller.api.devices[mac]
+        entities.append(UniFiDeviceUpdateEntity(device, controller))
+
+    if entities:
+        async_add_entities(entities)
+
+
+class UniFiDeviceUpdateEntity(UniFiBase, UpdateEntity):
+    """Update entity for a UniFi network infrastructure device."""
+
+    DOMAIN = DOMAIN
+    TYPE = DEVICE_UPDATE
+    _attr_device_class = UpdateDeviceClass.FIRMWARE
+    _attr_supported_features = UpdateEntityFeature.PROGRESS
+
+    def __init__(self, device, controller):
+        """Set up tracked device."""
+        super().__init__(device, controller)
+
+        self.device = self._item
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self.device.name or self.device.model
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this device."""
+        return f"{self.TYPE}-{self.device.mac}"
+
+    @property
+    def available(self) -> bool:
+        """Return if controller is available."""
+        return not self.device.disabled and self.controller.available
+
+    @property
+    def in_progress(self) -> bool:
+        """Update installation in progress."""
+        return self.device.state == 4
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+        return self.device.version
+
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        return self.device.upgrade_to_firmware or self.device.version
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+        info = DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, self.device.mac)},
+            manufacturer=ATTR_MANUFACTURER,
+            model=self.device.model,
+            sw_version=self.device.version,
+        )
+
+        if self.device.name:
+            info[ATTR_NAME] = self.device.name
+
+        return info
+
+    async def options_updated(self) -> None:
+        """No action needed."""
```

Comment:
Or should I remove the `NotImplementedError` at https://github.com/home-assistant/core/blob/49491bcda55125369319fe1f0b6036ddfc4a2a2a/homeassistant/components/unifi/unifi_entity_base.py#L76-L78 ?

Reply:
I will clean that up later on, this is a good reminder for now.